core/amber/src/main/scala/edu/uci/ics/texera/web/resource/auth/UserResource.scala

```diff
@@ -0,0 +1,103 @@
+package edu.uci.ics.texera.web.resource.auth
+
+import edu.uci.ics.texera.web.SqlServer
+import edu.uci.ics.texera.web.model.common.auth.User
+import edu.uci.ics.texera.web.model.jooq.generated.Tables.USER
+import edu.uci.ics.texera.web.model.request.auth.{UserLoginRequest, UserRegistrationRequest}
+import edu.uci.ics.texera.web.model.response.GenericWebResponse
+import edu.uci.ics.texera.web.model.response.auth.UserWebResponse
+import io.dropwizard.jersey.sessions.Session
+import javax.servlet.http.HttpSession
+import javax.ws.rs._
+import javax.ws.rs.core.MediaType
+import org.apache.commons.lang3.tuple.Pair
+import org.jooq.Condition
+import org.jooq.impl.DSL.defaultValue
+
+@Path("/users/")
+@Consumes(Array(MediaType.APPLICATION_JSON))
+@Produces(Array(MediaType.APPLICATION_JSON)) object UserResource {
+  private val SESSION_USER = "texera-user"
+
+  def getUser(session: HttpSession): User =
+    session.getAttribute(SESSION_USER).asInstanceOf[User]
+
+  private def setUser(session: HttpSession, user: User): Unit = {
+    session.setAttribute(SESSION_USER, user)
+  }
+
+}
+
+@Path("/users/")
+@Consumes(Array(MediaType.APPLICATION_JSON))
+@Produces(Array(MediaType.APPLICATION_JSON)) class UserResource {
```

Comment:
formatting seems off, make sure you use scalafmt, go to intellij settings and make sure scalafmt is used